device/sycl/src/clusterization/measurement_creation.sycl

```diff
@@ -0,0 +1,150 @@
+    // measurments seems not to have enough capacity which seems wrong
+    // because the number of measurments per module (the inner vector)
+    // cannot exceed the number of clusters per module
+    std::vector<std::size_t> cluster_sizes(cell_modules_per_event.size(), 1);
+    for (const auto &cl_id : clusters.get_headers()) {
+        cluster_sizes.at(cl_id.module_idx)++;
+    }
+
+    // Resizable buffer for the measurements
+    measurement_container_buffer measurement_buffer{
+        {size_buf, m_mr.get()},
+        {std::vector<std::size_t>(cluster_sizes.size(), 0), cluster_sizes,
+         m_mr.get()}};
+    copy.setup(measurement_buffer.items);
+
+    // Vector views to move data to the device
+    auto clusters_data = get_data(clusters, &m_mr.get());
+    cluster_container_const_view clusters_view(clusters_data);
+    measurement_container_view measurement_view(measurement_buffer);
+
+    // range of kernel execution
+    auto range = clusters.size();
+
+    // Run the kernel
+    details::get_queue(m_queue)
+        .parallel_for(
+            range,
+            [clusters_view, measurement_view](auto idx) {
+                // Initialize device vectors
+                device_cluster_const_container clusters_device(clusters_view);
+                device_measurement_container measurement_device(
+                    measurement_view);
+
+                // Ignore if idx is out of range
+                if (idx >= clusters_device.size())
+                    return;
+
+                // items: cluster of cells at current idx
+                // header: cluster_id object with the information about the cell
+                // module
+                const device_cell_const_collection &cluster =
+                    clusters_device.get_items().at(idx);
+                const cluster_id &cl_id = clusters_device.get_headers().at(idx);
+
+                const vector2 pitch = get_pitch(cl_id);
+                const auto module_idx = cl_id.module_idx;
+
+                scalar totalWeight = 0.;
+
+                // To calculate the mean and variance with high numerical
+                // stability we use a weighted variant of Welford's algorithm.
+                // This is a single-pass online algorithm that works well for
+                // large numbers of samples, as well as samples with very high
+                // values.
+                //
+                // To learn more about this algorithm please refer to:
+                // [1] https://doi.org/10.1080/00401706.1962.10490022
+                // [2] The Art of Computer Programming, Donald E. Knuth, second
+                //     edition, chapter 4.2.2.
+                point2 mean = {0., 0.}, var = {0., 0.};
+
+                // Should not happen
+                if (cluster.empty()) {
+                    return;
+                }
+
+                for (const auto &cell : cluster) {
+                    scalar weight =
```

Comment:
Let's write a helper function which can be shared by host and device algorithm like we did for the seeding algorithm. We don't have to repeat the same codes

Reply:
It's a good idea. Let's move the code from this for-loop into a separate helper function. Which you should also update the CPU code to use.

I'm thinking of something like:

```c++
TRACCC_HOST_DEVICE
void calc_cluster_properties(const cluster& cl, point2& mean, point2& variance,
                             scalar& totalWeight);
```

Reply:
I made a `measurement_creation_helper.hpp` inside core/clusterization where I put the functions that are reused for CPU and SYCL.

I was thinking at first of making a `traccc::common` library, similarly as you've done in `device/common` in #170. Would that be something to consider for the future?

Reply:
Not exactly. I should have been a little clearer... :frowning:

Please move the for-loop into the new function as well. Don't call that function from the same sort of for-loop in both the device and the host code. This is why I wrote that the function should receive an entire cluster, not just one cell.

Reply:
I templated the function to receive either `host_cell_collection` or `device_cell_collection`. For the device code, it would actually need to receive a `device_cell_const_collection`... but for now I "casted away" the const in SYCL code to make it work.

Reply:
> I was thinking at first of making a traccc::common library
IMAO, It should be enough to make `traccc/clusterization/common` directory and move helper files into it.